unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Transform/Handler/ECSTransformHandler.cs

```diff
@@ -93,6 +93,14 @@ private static void ProcessNewParent(IParcelScene scene, IDCLEntity entity, long
                 }
             }
 
+            //check for cyclic parenting
+            if (ECSTransformUtils.IsCircularParenting(scene, entity, parentId))
+            {
+                Debug.LogError($"cyclic parenting found for entity {entity.entityId} " +
+                               $"parenting to {parentId} at scene {scene.sceneData.id} ({scene.sceneData.basePosition})");
+                return;
+            }
```

Comment:
this should be called at the beginning of `private static void ProcessNewParent(IParcelScene scene, IDCLEntity entity, long parentId)` right? otherwise in the recursive parenting situation the entity would lose its previous parent and then be unparented

Reply:
good catch, fixed here: https://github.com/decentraland/unity-renderer/pull/2837/commits/e09ce0c5c3140ae9c74852f70ec4ab1cfd71ff0b